nvflare/private/fed/server/fed_server.py

```diff
@@ -687,12 +700,22 @@ def start_run(self, job_id, run_root, conf, args, snapshot):
                 if self.engine.asked_to_stop:
                     self.engine.engine_info.status = MachineStatus.STOPPED
 
+                self._send_parent_heartbeat(job_id)
                 time.sleep(self.check_engine_frequency)
 
         finally:
             self.engine.engine_info.status = MachineStatus.STOPPED
             self.run_manager = None
 
+    def _send_parent_heartbeat(self, job_id):
+        request = new_cell_message({CellMessageHeaderKeys.JOB_ID: job_id}, {})
+        self.cell.fire_and_forget(
```

Comment:
The cell may have not been created. Check to make sure self.cell is created before sending.

Reply:
The cell should have been created earlier before this. But added the check any way.